product-backend/src/main/java/org/kainos/ea/filter/AuthFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.kainos.ea.filter;
+
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.core.Response;
+
+import java.security.interfaces.RSAPublicKey;
+
+
+public class AuthFilter implements ContainerRequestFilter {
+    private final String secretKey = System.getenv("JWT_SECRET");
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) {
+        String path = requestContext.getUriInfo().getPath();
+        if (!path.contains("auth")) {
+            try {
+                String token = requestContext.getHeaderString("Authorization");
+                if (token == null) {
+                    throw new WebApplicationException("User unauthorized", Response.Status.UNAUTHORIZED);
+                }
+                DecodedJWT jwtData = JWT.decode(token);
+                if (path.contains("admin")) {
+                    if (jwtData.getClaim("roleId").asInt() != 2) {
+                        throw new WebApplicationException("No admin privileges", Response.Status.UNAUTHORIZED);
+                    }
+                }
```

Comment:
You are missing the key part of the authorisation - validating the token is valid and it didn't expire
https://github.com/auth0/java-jwt#verify-a-jwt